src/lib.rs

```diff
@@ -298,10 +298,12 @@ impl LoopDevice {
     ///
     /// This function needs to stat the backing file and can fail if there is
     /// an IO error.
+    #[allow(clippy::unnecessary_cast)]
     pub fn major(&self) -> io::Result<u32> {
         self.device
             .metadata()
-            .map(|m| unsafe { libc::major(m.rdev()) as u32 })
+            .map(|m| unsafe { libc::major(m.rdev()) })
+            .map(|m| m as u32)
```

Comment:
Why is this cast needed? If it is would it be better to use `.into()` which will error if the cast is ever going to possibly lose data.

Reply:
The bionic API differs here and yields a `i32` for major/minor. Unfortunately there's no `impl From<i32> for u32`.
The cast is nasty but safe because major/minors are basically `u8`.